packages/opentelemetry-exporter-zipkin/src/zipkin.ts

```diff
@@ -27,16 +27,17 @@ import {
   statusDescriptionTagName,
 } from './transform';
 import { OT_REQUEST_HEADER } from './utils';
+import { SERVICE_RESOURCE } from '@opentelemetry/resources';
 /**
  * Zipkin Exporter
  */
 export class ZipkinExporter implements SpanExporter {
   static readonly DEFAULT_URL = 'http://localhost:9411/api/v2/spans';
   private readonly _logger: api.Logger;
-  private readonly _serviceName: string;
   private readonly _statusCodeTagName: string;
   private readonly _statusDescriptionTagName: string;
   private readonly _reqOpts: http.RequestOptions;
+  private _serviceName: string;
```

Comment:
If this is not made optional, and not configured in the constructor after fixing the other comment, the types will complain because this is not definitely set in the constructor. Making it optional will fix this.

```suggestion
  private _serviceName?: string;
```

Reply:
doing this will cause an error when `_serviceName` is passed to `toZipkinSpan` method because service name can now be either `string` or `undefined`.

Reply:
Typescript is smart enough to handle the type assertion when combined with the `if`:

```typescript
someNumber: number = 0;
let x: number | undefined;

if (typeof x !== 'number') {
  x = someNumber;
}

// typescript is smart enough to know that type of x is number here

takesNumber(x);

function takesNumber(n: number) {
  // ...
}
```

Reply:
I have applied the changes in https://github.com/open-telemetry/opentelemetry-js/pull/1138#discussion_r434774628 and I have also made it optional in the config type. But since `this._serviceName` is being set in the `export` method, `toZipkinSpan` method still interprets `this._serviceName` as `string | undefined`.
here's the error:
`Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.`

Reply:
Ah I see. It is because it is used in another function so the typings are lost. You will have to modify the signature of that private function so that the service name is passed in explicitly as a string. Actually, it looks like a useless private method anyways, so maybe you should just inline the call to `toZipkinSpan`

Reply:
Or how about setting `serviceName` inside `_toZipkinSpan`?
Because spans are sent in order to that method we can use the resource service name from the first span that is sent. I've tried this and it works.

Reply:
It may work but it is non-obvious and it would be easy to break accidentally later without noticing. Having it explicitly as the first thing that happens on export makes it clear what is happening.